### ISSUE TITLE
Support prefetching of filters

### ISSUE THREAD

**Issue by klausenbusk** [uBlockOrigin/uBlock-issues#87]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

We us Firefox and uBlock as part of a digital signage solution, and Firefox is started from a clean profile on every boot (only `extensions/uBlock0@raymondhill.net.xpi` + `extensions.json`). We use the `adminSettings` [feature](https://github.com/gorhill/uBlock/wiki/Deploying-uBlock-Origin), to pre-enable a few filter lists (`.mozilla/managed-storage/uBlock0@raymondhill.net.json`), but we also need a way to pre-fetch them. Would it make sense to support that use-case? (For now I can workaround it by modifying `assets/assets.json` and adding the filters to `assets/foo/bar`). We need to pre-fetch them, as internet isn't always available when Firefox is started.

**Comment by gorhill** [comment]
I don't see how this is possible -- unless the extra filter lists are proxied through some local server which is never down.

**Comment by klausenbusk** [comment]
> I don't see how this is possible -- unless the extra filter lists are proxied through some local server which is never down.

For now I'm just using the following script:
```
prefetch_uBlock () {
	TMP="$(mktemp -d)"
	trap "rm -fr $TMP" EXIT

	EXT="${DIR}/firefox-profile/extensions/uBlock0@raymondhill.net.xpi"
	unzip "${EXT}" -d "${TMP}"
	mkdir "${TMP}/assets/prefetch"
	curl "https://secure.fanboy.co.nz/fanboy-cookiemonster.txt" -o "${TMP}/assets/prefetch/fanboy-cookiemonster.txt" -s -f
	jq '."fanboy-cookiemonster".contentURL |= [., "assets/prefetch/fanboy-cookiemonster.txt"]' < "${TMP}/assets/assets.json" > "${TMP}/assets/assets.json.tmp"
	mv "${TMP}/assets/assets.json"{.tmp,}
	(cd "${TMP}"; zip -r -FS "${EXT}" .)
}
```
As WebExtensions can't read local file, I can't come up with a better solution.

**Comment by gorhill** [comment]
> As WebExtensions can't read local file, I can't come up with a better solution.

Yes, still an open issue: <https://bugzilla.mozilla.org/show_bug.cgi?id=1266960>.

**Comment by gorhill** [comment]
Does the fix to <https://github.com/uBlockOrigin/uBlock-issues/issues/666> help?

**Comment by gorhill** [comment]
Given no feedback, I will assume fix to #666 which allow custom `assets.json` through `adminSettings.assetsBootstrapLocation` should take care of the issue here as well.
